tests/test_connector.py

```diff
@@ -56,7 +56,15 @@ def unix_server(loop, unix_sockname):
         runners.append(runner)
         await runner.setup()
         site = web.UnixSite(runner, unix_sockname)
-        await site.start()
+        try:
+            await site.start()
+        except OSError as exc:
+            if str(exc) == "AF_UNIX path too long":
+                if sys.platform == "linux":
+                    raise
+                pytest.skip("Some UNIXes has too strict limitation "
```

Comment:
It looks like a proper workaround it using `tmpdir_factory` fixture: https://github.com/agronholm/anyio/pull/42/files#diff-f0c1c7497545019b8489fa8bc06be6cdR208

Reply:
It is a workaround that doesn't guarantee short path anyway :(
Also, I prefer `tmp_path_factory` over `tmpdir_factory`

Reply:
Yep, using `tmp_path_factory` now.

Reply:
I've added a few improvements and it is now guaranteed to be short.